Uplift/Views/GymDetail/GymDetailTimeInfoView.swift

```diff
@@ -10,76 +10,92 @@ import UIKit
+    private let timesTextView = UITextView()
+    private let paragraphStyle = NSMutableParagraphStyle()
 
-    private var facilityDetail: FacilityDetail
-    private var selectedDayIndex = 0
     private var displayedText = ""
-    private var timesText = NSMutableAttributedString()
-    private var paragraphStyle = NSMutableParagraphStyle()
+    private var emptyStateView: UIView?
+    private var facilityDetail: FacilityDetail!
+    private var hours: [FacilityHoursRange] = []
     private var paragraphStyleAttributes: [NSAttributedString.Key: Any] = [:]
+    private var selectedDayIndex = 0
 
     // MARK: - Init
-    init(facilityDetail: FacilityDetail) {
-        self.facilityDetail = facilityDetail
+    override init(frame: CGRect) {
         super.init(frame: .zero)
 
         timesText.mutableString.setString(displayedText)
 
-        let paragraphStyle = NSMutableParagraphStyle()
         paragraphStyle.lineSpacing = 5.0
         paragraphStyle.alignment = .center
         paragraphStyleAttributes = [
-            .font: UIFont._16MontserratRegular!,
+            .font: UIFont._16MontserratLight!,
             .paragraphStyle: paragraphStyle,
             .foregroundColor: UIColor.primaryBlack
         ]
 
-        displayTextView.attributedText = timesText
-        displayTextView.backgroundColor = .primaryWhite
-        displayTextView.font = ._16MontserratRegular
-        displayTextView.textColor = .primaryBlack
-        displayTextView.isScrollEnabled = false
-        displayTextView.isEditable = false
-        displayTextView.contentInset = .zero
-        displayTextView.textContainerInset = .zero
+        timesTextView.attributedText = timesText
+        timesTextView.backgroundColor = .primaryWhite
+        timesTextView.font = ._16MontserratLight
+        timesTextView.isScrollEnabled = false
+        timesTextView.isEditable = false
+        timesTextView.isSelectable = false
+        timesTextView.textContainerInset = .zero
+        timesTextView.textContainer.lineFragmentPadding = 0
 
-        addSubview(displayTextView)
+        addSubview(timesTextView)
 
         setupConstraints()
     }
 
+    func configure(facilityDetail: FacilityDetail, dayIndex: Int, onChangeDay: ((CGFloat, Int) -> Void)?) {
+        self.facilityDetail = facilityDetail
+        self.onChangeDay = onChangeDay
+        self.selectedDayIndex = dayIndex
+
+        updateAppearance()
+    }
+
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
     func setupConstraints() {
-        displayTextView.snp.makeConstraints { make in
-            make.edges.equalToSuperview()
+        timesTextView.snp.makeConstraints { make in
+            make.center.equalToSuperview()
         }
     }
     // MARK: - Update
     private func updateAppearance() {
+        if let emptyStateView = emptyStateView {
+            emptyStateView.removeFromSuperview()
+        }
```

Comment:
you could use nil coalescing operator here: `emptyStateView?.removeFromSuperview()`